geomstats/geometry/rank_k_psd_matrices.py

```diff
@@ -0,0 +1,209 @@
+        belongs = gs.logical_and(
+            gs.logical_and(is_symmetric, is_semipositive), is_rankk
+        )
+        return belongs
+
+    def projection(self, point):
+        r"""Project a matrix to the space of PSD matrices of rank k.
+
+        First the symmetric part of point is computed,
+        then the matrix is multiplied by the I_\epsilon matrix.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n, n]
+            Matrix to project.
+
+        Returns
+        -------
+        projected: array-like, shape=[..., n, n]
+            PSD matrix rank k.
+        """
+
+        sym = Matrices.to_symmetric(point)
+        _, s, v = gs.linalg.svd(sym)
+        h = gs.matmul(Matrices.transpose(v), s[..., None] * v)
+        sym_proj = (sym + h) / 2
+        eigvals, eigvecs = gs.linalg.eigh(sym_proj)
+        i = gs.array([0] * (self.n - self.rank) + [2 * gs.atol] * self.rank)
+        regularized = gs.assignment(eigvals, 0, gs.arange((self.n - self.rank)),
+                                    axis=0) + i
+        reconstruction = gs.einsum("...ij,...j->...ij", eigvecs, regularized)
+
+        return Matrices.mul(reconstruction, Matrices.transpose(eigvecs))
+
+    def random_point(self, n_samples=1, bound=1.0):
+        r"""Sample in PSD(n,k) from the log-uniform distribution.
+
+        Parameters
+        ----------
+        n_samples : int
+            Number of samples.
+            Optional, default: 1.
+        bound : float
+            Bound of the interval in which to sample in the tangent space.
+            Optional, default: 1.
+
+        Returns
+        -------
+        samples : array-like, shape=[..., n, n]
+            Points sampled in PSD(n,k).
+        """
+        n = self.n
+        size = (n_samples, n, n) if n_samples != 1 else (n, n)
+        mat = bound * (2 * gs.random.rand(*size) - 1)
+        spd_mat = GeneralLinear.exp(Matrices.to_symmetric(mat))
+        return self.projection(spd_mat)
+
+    def is_tangent(self, vector, base_point):
+        r"""Check if the vector belongs to the tangent space.
+
+        Parameters
+        ----------
+        vector : array-like, shape=[..., n, n]
+            Matrix to check if it belongs to the tangent space.
+        base_point : array-like, shape=[..., n, n]
+            Base point of the tangent space.
+            Optional, default: None.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean denoting if vector belongs to tangent space
+            at base_point.
+        """
+        vector_sym = Matrices(self.n, self.n).to_symmetric(vector)
+
+        _, r = gs.linalg.eigh(base_point)
+        r_ort = r[..., :, self.n - self.rank: self.n]
+        r_ort_t = Matrices.transpose(r_ort)
+
+        candidates = gs.matmul(
+            gs.matmul(gs.matmul(r_ort, r_ort_t), vector_sym), gs.matmul(r_ort, r_ort_t)
+        )
```

Comment:
First you can use `Matrices.mul(a,b,c,d...)` instead of repeating the call to `matmut`. Than you can assigne `gs.matmul(r_ort, r_ort_t)` to a variable to avoid computing it twice.